packages/hydrogen/src/Numbers/UIStaticNumber.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { Text, StyleSheet } from 'react-native';
+
+import { uiLocalized } from '@tonlabs/uikit.localization';
+
+import { Typography } from '../Typography';
+import { useTheme, ColorVariants } from '../Colors';
+
+import { useNumberValue } from './useNumberValue';
+import {
+    runOnUILocalizedNumberFormat,
+    UINumberDecimalAspect,
+} from './runOnUILocalizedNumberFormat';
+import type { UINumberAppearance, UINumberGeneralProps } from './types';
+import { DebugGrid } from './DebugGrid';
+
+export function useNumberStaticStyles(integerColor: ColorVariants, decimalColor: ColorVariants) {
+    const theme = useTheme();
+    const integerColorStyle = React.useMemo(
+        () => ({ color: theme[integerColor] }),
+        [theme, integerColor],
+    );
+    const decimalColorStyle = React.useMemo(
+        () => ({ color: theme[decimalColor] }),
+        [theme, decimalColor],
+    );
+
+    return [integerColorStyle, decimalColorStyle];
+}
+
+export function UIStaticNumber({
+    testID,
+    children,
+    integerVariant,
+    integerColor,
+    decimalVariant,
+    decimalColor,
+    decimalAspect = UINumberDecimalAspect.None,
+    sign,
+    showDebugGrid,
+    showPositiveSign,
+}: UINumberGeneralProps & UINumberAppearance & { sign?: React.ReactNode }) {
+    const value = useNumberValue(children, decimalAspect);
+    const { decimal: decimalSeparator, grouping: integerGroupChar } =
+        uiLocalized.localeInfo.numbers;
+
+    const formatted = React.useMemo(() => {
+        return runOnUILocalizedNumberFormat(
+            value,
+            decimalAspect,
+            decimalSeparator,
+            integerGroupChar,
+            showPositiveSign,
+        );
+    }, [value, decimalAspect, decimalSeparator, integerGroupChar, showPositiveSign]);
```

Comment:
How about dealing with `BigNumber` here only, since we don't need to animate it using `withTiming`..